spec/i18n/placeholders_spec.rb

```diff
@@ -0,0 +1,45 @@
+describe "Placeholders in strings" do
+  it "translations preserve placeholders in strings" do
+    dirs = [Rails.root.join('locale')]
+
+    Rails::Engine.subclasses.each do |engine|
+      dir = engine.root.join('locale')
+      dirs << dir if dir.exist?
+    end
+
+    errors = {}
+
+    dirs.each do |dir|
+      Pathname.glob(::File.join(dir, "**", "*.po")).each do |po_file|
+        po = FastGettext::PoFile.new(po_file)
+        locale = po_file.dirname.basename.to_s
+        next if locale == 'en'
+        po.data.keys.each do |key|
+          next if po.data[key].nil?
+          placeholders = key.scan /%{\w+}/
+          placeholders.sort!
+          next if placeholders.empty?
+          translated_placeholders = po.data[key].scan /%{\w+}/
+          translated_placeholders.sort!
+          if placeholders.uniq != translated_placeholders.uniq
+            errors[po_file.to_s] ||= []
+            errors[po_file.to_s] << [key, po.data[key]]
```

Comment:
Since you are pushing a key, value pair, this makes me think you should just use nested Hashes and store it with store_path.  So, line 10 above would just be `errors = {}`, you can remove line 25, and this line would just be `errors.store_path(po_file.to_s, key, po.data[key])`.